### PR TITLE
Add Azure AD authentication

### PR DESCRIPTION
**Pull request by sschritt**
Authentication with Microsoft AzureAD SSO (see zulip/zulip#10618)

Fixes: #3225

### CODE REVIEW

File: src/common/Icons.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ export const IconPrivateChat = (props: Object) => <Feather name="mail" {...props
 export const IconDownArrow = (props: Object) => <Feather name="chevron-down" {...props} />;
 export const IconGoogle = (props: Object) => <IoniconsIcon name="logo-google" {...props} />;
 export const IconGitHub = (props: Object) => <Feather name="github" {...props} />;
+export const IconAzureAD = (props: Object) => <IoniconsIcon name="logo-windows" {...props} />;
```
Comment (borisyankov) [inline src/common/Icons.js]: Maybe call it `IconWindows` even though we use it for Azure.
The logo for Azure is different, but I think Windows' logo is a reasonable replacement.

**Comment by borisyankov** [comment]
@sschritt the PR looks good. Would you resolve the merge conflicts?
I think we want to merge it ASAP as discussed here https://chat.zulip.org/#narrow/stream/48-mobile/topic/Azure.20SSO

We don't currently have a server with that authentication set up to test on but I have high confidence the code is correct and should work.
We can merge it as-is.

**Comment by borisyankov** [comment]
@gnprice I did add a P1 to make the PR more visible.
It is solving a problem a customer is encountering.

**Comment by gnprice** [comment]
@borisyankov I just went and did the rebase -- the conflicts were straightforward to resolve. Also squashed the name change into the main commit.

**Comment by gnprice** [comment]
Two things that need to be fixed:
* A test is broken.
* The `AuthenticationMethods` type should be updated to include the new case we're acting on.

Those should be easy, so I'll just fix them and merge.

**Comment by gnprice** [comment]
Merged! @sschritt , please try it out and confirm everything works for you.

**Comment by Vilinthril** [comment]
Just to clarify, this has not been promulgated to the Google Play Store yet, right?

**Pull request merged**